### PR TITLE
[4138] Language and currency reset on every page load in studio preview

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4138

### CODE REVIEW

File: sources/ecommerce-app/src/util/component.js [hunk 1]
```diff
@@ -50,16 +51,19 @@ export function useHeader() {
...
     () => {
-      if (cart == null) {
-        dispatch(fetchCart());
-      } else if (cart.items.length === 0) {
+      if (cart && cart.items.length === 0) {
         onEmpty && onEmpty();
       }
     },
-    [cart]
+    [cart, onEmpty]
```
Comment (rart) [inline sources/ecommerce-app/src/util/component.js:62]: Is this safe? Are consumers of useCart memoizing onEmpty?

File: sources/ecommerce-app/src/util/component.js [hunk 2]
```diff
@@ -75,7 +79,7 @@ export function useUser({ redirect, onMissing } = {}) {
           onMissing && onMissing();
       }
     },
-    []
+    [dispatch, user, onMissing, redirect]
```
Comment (rart) [inline sources/ecommerce-app/src/util/component.js:82]: Is this safe? Are consumers of useUser memoizing redirect & onMissing?

File: sources/ecommerce-app/src/util/component.js [hunk 3]
```diff
@@ -92,7 +96,7 @@ export function useOrder(id) {
       if (id && (order == null || order.items == null))
         dispatch(fetchOrder(id));
     },
-    [id]
+    [dispatch, id, order]
```
Comment (rart) [inline sources/ecommerce-app/src/util/component.js:99]: Sage? (order)

File: sources/ecommerce-app/src/util/component.js (hunk 1 above)
Comment (aachavarria) [reply sources/ecommerce-app/src/util/component.js:62]: using refs

File: sources/ecommerce-app/src/util/component.js (hunk 2 above)
Comment (aachavarria) [reply sources/ecommerce-app/src/util/component.js:82]: using refs

File: sources/ecommerce-app/src/util/component.js (hunk 3 above)
Comment (aachavarria) [reply sources/ecommerce-app/src/util/component.js:99]: yep

**Pull request merged**